server/events/yaml/raw/global_cfg.go

```diff
@@ -174,17 +187,32 @@ func (r Repo) ToValid(workflows map[string]valid.Workflow) valid.Repo {
 		workflow = &ptr
 	}
 
-	preWorkflowHooks := make([]*valid.PreWorkflowHook, 0)
+	var preWorkflowHooks []*valid.PreWorkflowHook
 	if len(r.PreWorkflowHooks) > 0 {
 		for _, hook := range r.PreWorkflowHooks {
 			preWorkflowHooks = append(preWorkflowHooks, hook.ToValid())
 		}
 	}
 
+	var mergedApplyReqs []string
+
+	mergedApplyReqs = append(mergedApplyReqs, r.ApplyRequirements...)
+
+	// only add global reqs if they don't exist already.
+OUTER:
+	for _, globalReq := range globalApplyReqs {
+		for _, currReq := range r.ApplyRequirements {
+			if globalReq == currReq {
+				continue OUTER
```

Comment:
I'm trying to understand this part, doesn't this create infinite loop? This is basically a GOTO statement right?

Reply:
hmm, no this just allows us to continue from the next iteration of the outer loop without all the ugly syntax of having a boolean to flag this and then checking that in the outer loop before appending the requirement to the result list.